### PR TITLE
Desktop: Setup SonarCloud for the CI

### PR DESCRIPTION
**Pull request by FlexW**
Signed-off-by: Felix Weilbach <felix.weilbach@nextcloud.com>

### CODE REVIEW

**Comment by FlexW** [comment]
Will be useful for https://github.com/nextcloud/desktop/pull/3861 to prevent extra work each time.

File: client/Dockerfile [hunk 1]
```diff
@@ -180,3 +182,22 @@ RUN if [ "$BUILD_QT" != "1" ] ; then \
...
+ENV SONAR_SCANNER_DOWNLOAD_URL https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${SONAR_SCANNER_VERSION}-linux.zip
+
+RUN mkdir -p /opt/sonar && \
+    curl -sSLo /opt/sonar/sonar-scanner.zip ${SONAR_SCANNER_DOWNLOAD_URL} && \
+    unzip -o /opt/sonar/sonar-scanner.zip -d /opt/sonar/
+
+ENV PATH=/opt/sonar/sonar-scanner-${SONAR_SCANNER_VERSION}-linux/bin:${PATH}
+
+# Setup build-wrapper
+ENV BUILD_WRAPPER_DOWNLOAD_URL ${SONAR_SERVER_URL}/static/cpp/build-wrapper-linux-x86.zip
```
Comment (mgallien) [inline client/Dockerfile:198]: why is it `x86` and not `amd64` or `x86_64` ?

Comment (FlexW) [reply client/Dockerfile:198]: Good question, but that is from the official guide, and I don't see that other versions exist.

Comment (mgallien) [reply client/Dockerfile:198]: OK, can you add a pointer to the doc here for later reference ?

**Review by mgallien — APPROVED** [review]

**Pull request merged**